extern/fastor/CMakeLists.txt

```diff
@@ -15,16 +15,36 @@ message( STATUS "Building Fastor as part of the Algebra Plugins project" )
 # We need to use this alternative syntax for FetchContent_Declare because the
 # latest release for Fastor does not have a CMakeLists.txt file.
 set( ALGEBRA_PLUGINS_FASTOR_SOURCE
-   "fastor;GIT_REPOSITORY;https://github.com/romeric/Fastor.git;GIT_TAG;6d7216f304d60d92b9b042566d4d6e94c65086a3"
+   "fastor;GIT_REPOSITORY;https://github.com/romeric/Fastor.git;GIT_TAG;e96e63fe8a05898d732d0b9faba3667d9a25a38f"
    CACHE STRING "Source for Fastor, when built as part of this project" )
 mark_as_advanced( ALGEBRA_PLUGINS_FASTOR_SOURCE )
 FetchContent_Declare( fastor ${ALGEBRA_PLUGINS_FASTOR_SOURCE} )
 
-# Options used in the build of Fastor.
-set(BUILD_TESTING OFF CACHE BOOL "Don't build the Fastor tests")
-add_compile_definitions(FASTOR_ENABLE_RUNTIME_CHECKS=0)
+# Turn off the unit tests for Fastor.
+if( DEFINED CACHE{BUILD_TESTING} )
+   set( _buildTestingValue ${BUILD_TESTING} )
+endif()
+set( BUILD_TESTING FALSE CACHE INTERNAL "Forceful setting of BUILD_TESTING" )
 
 # Get it into the current directory.
-FetchContent_Populate( fastor )
-add_subdirectory( "${fastor_SOURCE_DIR}" "${fastor_BINARY_DIR}"
-   EXCLUDE_FROM_ALL )
+FetchContent_MakeAvailable( fastor )
+
+# Reset the BUILD_TESTING variable.
+if( DEFINED _buildTestingValue )
```

Comment:
We _are_ turning off the unit tests in Fastor.

What we are doing is caching the current value of `BUILD_TESTING` in [this portion of the code](https://github.com/wermos/algebra-plugins/blob/fcfde61b28f81ddce945a054f404a1928db7058a/extern/fastor/CMakeLists.txt#L24-L27), turning off the Fastor unit tests, and then [setting `BUILD_TESTING` back to its original value](https://github.com/wermos/algebra-plugins/blob/fcfde61b28f81ddce945a054f404a1928db7058a/extern/fastor/CMakeLists.txt#L32-L39). The reason we do it like this is because we want to turn off tests for Fastor specifically but not tamper with the value of `BUILD_TESTING` in the parent repository (`algebra-plugins`) in case someone wants to run the `algebra-plugins` tests.